### ISSUE TITLE
About setting special tokens when fine-tuning

### ISSUE THREAD

**Issue by Remixa-python** [fate-ubw/RAGLAB#3]
I tried to find out how special tokens are handled in finetune. I found that in some operations, the label of special tokens is set to -100, so will special tokens not be updated?
like this: https://github.com/fate-ubw/RAGLAB/blob/main/raglab/rag/train_alg/finetune.py#L286
If my understanding is wrong, please point it out for me. Thank you very much for your contribution!

**Comment by fate-ubw** [comment]
- I'm very pleased that you're using RAGLAB, and I'm grateful for you pointing out the issue! Regarding the training of special tokens, there seems to be a slight misunderstanding in your interpretation. Let me explain the special token-related issues:

- First, let me clarify the concept of special tokens. Special tokens originate from the SelfRAG paper, which proposed adding extra tokens during the fine-tuning phase. These extra tokens have their own specific functions. However, using the term "special token" in SelfRAG might mislead some beginners. This is because during the training phase, special tokens participate in training just like ordinary tokens. The training process in SelfRAG is essentially no different from expanding the vocabulary. For example, when LLaMA3-Chinese adds more Chinese vocabulary during fine-tuning, it enables the trained model to generate new vocabulary. SelfRAG's training process is consistent with this approach. Special tokens are not masked any differently from other ordinary tokens.

- So why does this line of code appearlabels[start_idx+1:end_idx] = -100？

- In fact, this masking operation is unrelated to special tokens. This masking operation is unique to the SelfRAG model. Because the SelfRAG model needs to answer questions based on retrieved passages during inference, the training process needs to make the model learn from the retrieved passages. However, during inference, the model should only output the answer, not the retrieved passages. To achieve this goal, it's necessary to mask the retrieved passage text segments during training. This is the main function of this part of the code. The original paper expresses this issue as follows:

<img width="1102" alt="train" src="https://github.com/user-attachments/assets/f9900c1a-dd6c-4149-95e5-5259e0f86e61">

- I hope this answers your question. We are in the process of building clearer code logic, and we welcome more questions.

**Comment by Remixa-python** [comment]
@fate-ubw 
Thank you for your patience! I get it! 
This has nothing to do with special tokens, but is only for SelfRAG's use case.

**Comment by fate-ubw** [comment]
I'm going to close this issue and feel free to reopen it～🤗
